src/ParallelAlgorithms/LinearSolver/Schwarz/OverlapHelpers.cpp

```diff
@@ -0,0 +1,75 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "ParallelAlgorithms/LinearSolver/Schwarz/OverlapHelpers.hpp"
+
+#include <cmath>
+#include <cstddef>
+#include <string>
+
+#include "DataStructures/DataVector.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Utilities/EqualWithinRoundoff.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+
+namespace LinearSolver::Schwarz {
+
+size_t overlap_extent(const size_t volume_extent,
+                      const size_t max_overlap) noexcept {
+  return std::min(max_overlap, volume_extent - 1);
+}
+
+namespace {
+template <size_t Dim>
+void assert_overlap_extent(const Index<Dim>& volume_extents,
```

Comment:
why factor these `ASSERT`s when they're used only in one place?

Reply:
I'm going to use them in #2361 once I rebase that PR on top of this one

Reply:
Sounds good, thanks!